test/e2e/generate_incident.go

```diff
@@ -0,0 +1,201 @@
+package osde2etests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type PagerDutyClient interface {
+	CreateSilentRequest(alertName, clusterID string) (string, error)
+	GetIncidentID(dedupKey string) (string, error)
+	ResolveIncident(incidentID string) error
+}
+
+type client struct {
+	eventsURL     string
+	apiURL        string
+	routingKey    string
+	authToken     string
+	alertMappings map[string]string
+	httpClient    *http.Client
+}
+
+func NewClient(routingKey, authToken string) PagerDutyClient {
+	return &client{
+		eventsURL:  "https://events.pagerduty.com/v2/enqueue",
+		apiURL:     "https://api.pagerduty.com/incidents",
+		routingKey: routingKey,
+		authToken:  authToken,
+		alertMappings: map[string]string{
+			"ClusterHasGoneMissing":                         "cadtest has gone missing",
+			"ClusterProvisioningDelay":                      "ClusterProvisioningDelay -",
+			"ClusterMonitoringErrorBudgetBurnSRE":           "ClusterMonitoringErrorBudgetBurnSRE Critical (1)",
+			"InsightsOperatorDown":                          "InsightsOperatorDown",
+			"MachineHealthCheckUnterminatedShortCircuitSRE": "MachineHealthCheckUnterminatedShortCircuitSRE CRITICAL (1)",
+			"ApiErrorBudgetBurn":                            "api-ErrorBudgetBurn k8sgpt test CRITICAL (1)",
+		},
+		httpClient: &http.Client{Timeout: 10 * time.Second},
+	}
+}
+
+type payload struct {
+	Payload struct {
+		Summary   string            `json:"summary"`
+		Timestamp string            `json:"timestamp"`
+		Severity  string            `json:"severity"`
+		Source    string            `json:"source"`
+		Details   map[string]string `json:"custom_details"`
+	} `json:"payload"`
+	RoutingKey  string `json:"routing_key"`
+	EventAction string `json:"event_action"`
+	DedupKey    string `json:"dedup_key"`
+}
```

Comment:
There is a pagerduty go sdk that can be used, so we don't have to manually re-create structs and api calls here and in the other functions: https://github.com/PagerDuty/go-pagerduty